### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by wrandall22**
Based logic off of https://docs.google.com/a/cru.org/spreadsheets/d/1w-TwMusqsCOMeyhZTjuiVRfh3JlbfrHzY5dssA7KVkw/edit#gid=0

### CODE REVIEW

**Comment by wrandall22** [comment]
@frett @mattfred

**Comment by wrandall22** [comment]
Also based off of https://github.com/GlobalTechnology/gp_SingleBestRecord/wiki/Search-Churches

**Comment by wrandall22** [comment]
Adding authorization checks for editing training and churches now

**Comment by frett** [comment]
I'll wait until those auth checks are present before I review. I would say toggle the editable/enabled state of the views if the user isn't authorized.

**Comment by wrandall22** [comment]
Ok, not sure exactly what you meant by toggling the view states, but I put a role check in with the existing if statements.

**Comment by frett** [comment]
yup, implementing it right now in Edit Churches :)

**Comment by frett** [comment]
well, I guess it's not quick and easy to make an EditText view un-editable. So, I'll open a ticket with that enhancement and go with your approach for now :)

File: app/src/main/java/com/expidev/gcmapp/MainActivity.java [hunk 1]
```diff
@@ -589,13 +601,33 @@ private void addChurchMarkersToMap() {
...
+            return false;
+        }
+
+        switch(churchSecurity)
+        {
+            case 0:
+                return mAssignment.isLeader() || mAssignment.isMember();
+            case 1:
+                return mAssignment.isLeadership();
+            case 2:
```
Comment (frett) [inline app/src/main/java/com/expidev/gcmapp/MainActivity.java]: there is a church security of 3 from legacy data that should be treated as 2.

Long term church security will be converted to an enum which will capture this legacy data value.

**Comment by frett** [comment]
capture church security 3 and this is good

**Comment by wrandall22** [comment]
is that correct for 3?

**Comment by frett** [comment]
looks good

**Pull request merged**